File: fixtures/scale/elixir/small/lib/output/writer.ex
defmodule Output.Writer do
  alias Render.Box
  alias Style.Palette

  def render(palette, w, doc) do
    Enum.map(Box.draw(:rounded, w, doc), &Palette.pick(palette, &1))
  end

  def emit(lines), do: Enum.each(lines, &IO.puts/1)
end

File: fixtures/scale/haskell/small/Output/Writer.hs
module Output.Writer (render, emit) where

import Model.Document (Document)
import Render.Box (draw)
import Style.Border (Border(..))
import Style.Palette (Palette, pick)

render :: Palette -> Int -> Document -> [String]
render pal w doc = map (pick pal) (draw Rounded w doc)

emit :: [String] -> IO ()
emit = mapM_ putStrLn
